src/main/java/nl/tudelft/contextproject/gui/CreateScriptController.java

```diff
@@ -33,140 +49,380 @@
+ * script.
+ * 
+ * <p>The view section is defined under view/CreateScriptView.fxml
  * 
  * @since 0.2
  */
 public class CreateScriptController {
 
     private static boolean fill = false;
-    private Script script;
+
+    private ObjectProperty<TableRow<Shot>> lastSelectedRow;
+    private List<Shot> backupList;
+    private int maximumId = 0;
 
     @FXML private Button btnAdd;
     @FXML private Button btnBack;
+    @FXML private Button btnEditConfirm;
+    @FXML private Button btnEditRemove;
     @FXML private Button btnSave;
+    @FXML private Button btnSaveAs;
 
     @FXML private ChoiceBox<Number> addCamera;
-    @FXML private ChoiceBox<Number> addPreset;
+    @FXML private ChoiceBox<String> addPreset;
+    @FXML private ChoiceBox<Number> editCamera;
+    @FXML private ChoiceBox<String> editPreset;
 
     @FXML private HBox editBox;
 
+    @FXML private Pane editBlockScroll;
+
     @FXML private TableView<Shot> tableEvents;
     @FXML private TableColumn<Shot, Shot> columnAction;
     @FXML private TableColumn<Shot, Number> columnCamera;
     @FXML private TableColumn<Shot, String> columnDescription;
     @FXML private TableColumn<Shot, Number> columnID;
-    @FXML private TableColumn<Shot, Number> columnPreset;
+    @FXML private TableColumn<Shot, String> columnPreset;
+    @FXML private TableColumn<Shot, Image> columnReorder;
     @FXML private TableColumn<Shot, String> columnShot;
 
     @FXML private TextField addShot;
     @FXML private TextField addDescription;
+    @FXML private TextField editShot;
+    @FXML private TextField editDescription;
 
     /**
      * Initialize method used by JavaFX.
      */
     @FXML private void initialize() {
+        
+        // Set the current script as backup.
+        backupList = new ArrayList<Shot>();
+        backupList.addAll(0, ContextTFP.getScript().getShots());
+        
         setFactories();
         setAddButton();
         setBackButton();
         setSaveButton();
 
         initCamera();
-        initPreset();
+        initPreset("add");
+        initPreset("pres");
 
         allowEditing();
+        allowRowReordering();
 
         // Disallow horizontal scrolling.
         tableEvents.setColumnResizePolicy(TableView.CONSTRAINED_RESIZE_POLICY);
 
         // Allows the user to press ENTER to add a shot.
         btnAdd.setDefaultButton(true);
 
+        // Removes the "No content in table" label.
+        tableEvents.setPlaceholder(new Label(""));
+
         if (fill) {
             fillTable(ContextTFP.getScript().getShots());
             fill = false;
         }
         
-        // Store the current script locally to reduce traffic.
-        script = ContextTFP.getScript();
+        // Makes sure there are no duplicate ID's of the shots in the table.
+        for (Shot s : tableEvents.getItems()) {
+            if (s.getNumber() > maximumId) {
+                maximumId = s.getNumber();
+            }
+        }
```

Comment:
It does not necessarily check for duplicate IDs, but it notes the highest ID currently in use. Since every shot adds 1 to the ID, this will prevent IDs from being duplicated.